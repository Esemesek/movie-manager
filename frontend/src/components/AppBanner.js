import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { AppBar, Toolbar, Typography, Button } from '@material-ui/core';
import { ArrowBack } from '@material-ui/icons';

const renderBackButton = (goBack, history) => {
  if (!goBack) {
    return null;
  }

  return (
    <Button onClick={history.goBack}>
      <ArrowBack />
    </Button>
  );
}

export const AppBannerContainer = ({ goBack, history }) => (
  <AppBar color="default" position="static">
    <Toolbar>
      {renderBackButton(goBack, history)}
      <Typography variant="title">
        Movie Manager
      </Typography>
    </Toolbar>
  </AppBar>
);

AppBannerContainer.propTypes = {
  goBack: PropTypes.bool,
  history: PropTypes.instanceOf(Object).isRequired,
};

AppBannerContainer.defaultProps = {
  goBack: false,
};

export default withRouter(AppBannerContainer);
