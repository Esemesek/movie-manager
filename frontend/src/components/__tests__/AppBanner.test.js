import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { AppBannerContainer } from '../AppBanner';

const goBackMock = jest.fn();

const history = {
  goBack: goBackMock,
};

it('should render correctly with goBack option', () => {
  const wrapper = mount(<AppBannerContainer goBack history={history} />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('should render correctly without goBack button', () => {
  const wrapper = mount(<AppBannerContainer history={history} />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
