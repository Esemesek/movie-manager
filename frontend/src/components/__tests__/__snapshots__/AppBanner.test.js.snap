// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with goBack option 1`] = `
<Component
  goBack={true}
  history={
    Object {
      "goBack": [Function],
    }
  }
>
  <WithStyles(AppBar)
    color="default"
    position="static"
  >
    <AppBar
      classes={
        Object {
          "colorDefault": "MuiAppBar-colorDefault-6",
          "colorPrimary": "MuiAppBar-colorPrimary-7",
          "colorSecondary": "MuiAppBar-colorSecondary-8",
          "positionAbsolute": "MuiAppBar-positionAbsolute-3",
          "positionFixed": "MuiAppBar-positionFixed-2",
          "positionStatic": "MuiAppBar-positionStatic-5",
          "positionSticky": "MuiAppBar-positionSticky-4",
          "root": "MuiAppBar-root-1",
        }
      }
      color="default"
      position="static"
    >
      <WithStyles(Paper)
        className="MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
        component="header"
        elevation={4}
        square={true}
      >
        <Paper
          className="MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0-11",
              "elevation1": "MuiPaper-elevation1-12",
              "elevation10": "MuiPaper-elevation10-21",
              "elevation11": "MuiPaper-elevation11-22",
              "elevation12": "MuiPaper-elevation12-23",
              "elevation13": "MuiPaper-elevation13-24",
              "elevation14": "MuiPaper-elevation14-25",
              "elevation15": "MuiPaper-elevation15-26",
              "elevation16": "MuiPaper-elevation16-27",
              "elevation17": "MuiPaper-elevation17-28",
              "elevation18": "MuiPaper-elevation18-29",
              "elevation19": "MuiPaper-elevation19-30",
              "elevation2": "MuiPaper-elevation2-13",
              "elevation20": "MuiPaper-elevation20-31",
              "elevation21": "MuiPaper-elevation21-32",
              "elevation22": "MuiPaper-elevation22-33",
              "elevation23": "MuiPaper-elevation23-34",
              "elevation24": "MuiPaper-elevation24-35",
              "elevation3": "MuiPaper-elevation3-14",
              "elevation4": "MuiPaper-elevation4-15",
              "elevation5": "MuiPaper-elevation5-16",
              "elevation6": "MuiPaper-elevation6-17",
              "elevation7": "MuiPaper-elevation7-18",
              "elevation8": "MuiPaper-elevation8-19",
              "elevation9": "MuiPaper-elevation9-20",
              "root": "MuiPaper-root-9",
              "rounded": "MuiPaper-rounded-10",
            }
          }
          component="header"
          elevation={4}
          square={true}
        >
          <header
            className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
          >
            <WithStyles(Toolbar)>
              <Toolbar
                classes={
                  Object {
                    "gutters": "MuiToolbar-gutters-37",
                    "root": "MuiToolbar-root-36",
                  }
                }
                disableGutters={false}
              >
                <div
                  className="MuiToolbar-root-36 MuiToolbar-gutters-37"
                >
                  <WithStyles(Button)
                    onClick={[Function]}
                  >
                    <Button
                      classes={
                        Object {
                          "colorInherit": "MuiButton-colorInherit-43",
                          "disabled": "MuiButton-disabled-48",
                          "fab": "MuiButton-fab-49",
                          "flatPrimary": "MuiButton-flatPrimary-40",
                          "flatSecondary": "MuiButton-flatSecondary-41",
                          "focusVisible": "MuiButton-focusVisible-47",
                          "fullWidth": "MuiButton-fullWidth-53",
                          "label": "MuiButton-label-39",
                          "mini": "MuiButton-mini-50",
                          "outlined": "MuiButton-outlined-42",
                          "raised": "MuiButton-raised-44",
                          "raisedPrimary": "MuiButton-raisedPrimary-45",
                          "raisedSecondary": "MuiButton-raisedSecondary-46",
                          "root": "MuiButton-root-38",
                          "sizeLarge": "MuiButton-sizeLarge-52",
                          "sizeSmall": "MuiButton-sizeSmall-51",
                        }
                      }
                      color="default"
                      disableFocusRipple={false}
                      disabled={false}
                      fullWidth={false}
                      mini={false}
                      onClick={[Function]}
                      size="medium"
                      type="button"
                      variant="flat"
                    >
                      <WithStyles(ButtonBase)
                        className="MuiButton-root-38"
                        disabled={false}
                        focusRipple={true}
                        focusVisibleClassName="MuiButton-focusVisible-47"
                        onClick={[Function]}
                        type="button"
                      >
                        <ButtonBase
                          centerRipple={false}
                          className="MuiButton-root-38"
                          classes={
                            Object {
                              "disabled": "MuiButtonBase-disabled-55",
                              "focusVisible": "MuiButtonBase-focusVisible-56",
                              "root": "MuiButtonBase-root-54",
                            }
                          }
                          disableRipple={false}
                          disabled={false}
                          focusRipple={true}
                          focusVisibleClassName="MuiButton-focusVisible-47"
                          onClick={[Function]}
                          tabIndex="0"
                          type="button"
                        >
                          <button
                            className="MuiButtonBase-root-54 MuiButton-root-38"
                            disabled={false}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex="0"
                            type="button"
                          >
                            <span
                              className="MuiButton-label-39"
                            >
                              <pure(ArrowBack)>
                                <ArrowBack>
                                  <WithStyles(SvgIcon)>
                                    <SvgIcon
                                      classes={
                                        Object {
                                          "colorAction": "MuiSvgIcon-colorAction-60",
                                          "colorDisabled": "MuiSvgIcon-colorDisabled-62",
                                          "colorError": "MuiSvgIcon-colorError-61",
                                          "colorPrimary": "MuiSvgIcon-colorPrimary-58",
                                          "colorSecondary": "MuiSvgIcon-colorSecondary-59",
                                          "root": "MuiSvgIcon-root-57",
                                        }
                                      }
                                      color="inherit"
                                      viewBox="0 0 24 24"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        className="MuiSvgIcon-root-57"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <g>
                                          <path
                                            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                                          />
                                        </g>
                                      </svg>
                                    </SvgIcon>
                                  </WithStyles(SvgIcon)>
                                </ArrowBack>
                              </pure(ArrowBack)>
                            </span>
                            <WithStyles(TouchRipple)
                              center={false}
                              innerRef={[Function]}
                            >
                              <TouchRipple
                                center={false}
                                classes={
                                  Object {
                                    "child": "MuiTouchRipple-child-67",
                                    "childLeaving": "MuiTouchRipple-childLeaving-68",
                                    "childPulsate": "MuiTouchRipple-childPulsate-69",
                                    "ripple": "MuiTouchRipple-ripple-64",
                                    "ripplePulsate": "MuiTouchRipple-ripplePulsate-66",
                                    "rippleVisible": "MuiTouchRipple-rippleVisible-65",
                                    "root": "MuiTouchRipple-root-63",
                                  }
                                }
                              >
                                <TransitionGroup
                                  childFactory={[Function]}
                                  className="MuiTouchRipple-root-63"
                                  component="span"
                                  enter={true}
                                  exit={true}
                                >
                                  <span
                                    className="MuiTouchRipple-root-63"
                                  />
                                </TransitionGroup>
                              </TouchRipple>
                            </WithStyles(TouchRipple)>
                          </button>
                        </ButtonBase>
                      </WithStyles(ButtonBase)>
                    </Button>
                  </WithStyles(Button)>
                  <WithStyles(Typography)
                    variant="title"
                  >
                    <Typography
                      align="inherit"
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter-83",
                          "alignJustify": "MuiTypography-alignJustify-85",
                          "alignLeft": "MuiTypography-alignLeft-82",
                          "alignRight": "MuiTypography-alignRight-84",
                          "body1": "MuiTypography-body1-79",
                          "body2": "MuiTypography-body2-78",
                          "button": "MuiTypography-button-81",
                          "caption": "MuiTypography-caption-80",
                          "colorError": "MuiTypography-colorError-93",
                          "colorInherit": "MuiTypography-colorInherit-89",
                          "colorPrimary": "MuiTypography-colorPrimary-90",
                          "colorSecondary": "MuiTypography-colorSecondary-91",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary-92",
                          "display1": "MuiTypography-display1-74",
                          "display2": "MuiTypography-display2-73",
                          "display3": "MuiTypography-display3-72",
                          "display4": "MuiTypography-display4-71",
                          "gutterBottom": "MuiTypography-gutterBottom-87",
                          "headline": "MuiTypography-headline-75",
                          "noWrap": "MuiTypography-noWrap-86",
                          "paragraph": "MuiTypography-paragraph-88",
                          "root": "MuiTypography-root-70",
                          "subheading": "MuiTypography-subheading-77",
                          "title": "MuiTypography-title-76",
                        }
                      }
                      color="default"
                      gutterBottom={false}
                      headlineMapping={
                        Object {
                          "body1": "p",
                          "body2": "aside",
                          "display1": "h1",
                          "display2": "h1",
                          "display3": "h1",
                          "display4": "h1",
                          "headline": "h1",
                          "subheading": "h3",
                          "title": "h2",
                        }
                      }
                      noWrap={false}
                      paragraph={false}
                      variant="title"
                    >
                      <h2
                        className="MuiTypography-root-70 MuiTypography-title-76"
                      >
                        Movie Manager
                      </h2>
                    </Typography>
                  </WithStyles(Typography)>
                </div>
              </Toolbar>
            </WithStyles(Toolbar)>
          </header>
        </Paper>
      </WithStyles(Paper)>
    </AppBar>
  </WithStyles(AppBar)>
</Component>
`;

exports[`should render correctly without goBack button 1`] = `
<Component
  goBack={false}
  history={
    Object {
      "goBack": [Function],
    }
  }
>
  <WithStyles(AppBar)
    color="default"
    position="static"
  >
    <AppBar
      classes={
        Object {
          "colorDefault": "MuiAppBar-colorDefault-6",
          "colorPrimary": "MuiAppBar-colorPrimary-7",
          "colorSecondary": "MuiAppBar-colorSecondary-8",
          "positionAbsolute": "MuiAppBar-positionAbsolute-3",
          "positionFixed": "MuiAppBar-positionFixed-2",
          "positionStatic": "MuiAppBar-positionStatic-5",
          "positionSticky": "MuiAppBar-positionSticky-4",
          "root": "MuiAppBar-root-1",
        }
      }
      color="default"
      position="static"
    >
      <WithStyles(Paper)
        className="MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
        component="header"
        elevation={4}
        square={true}
      >
        <Paper
          className="MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
          classes={
            Object {
              "elevation0": "MuiPaper-elevation0-11",
              "elevation1": "MuiPaper-elevation1-12",
              "elevation10": "MuiPaper-elevation10-21",
              "elevation11": "MuiPaper-elevation11-22",
              "elevation12": "MuiPaper-elevation12-23",
              "elevation13": "MuiPaper-elevation13-24",
              "elevation14": "MuiPaper-elevation14-25",
              "elevation15": "MuiPaper-elevation15-26",
              "elevation16": "MuiPaper-elevation16-27",
              "elevation17": "MuiPaper-elevation17-28",
              "elevation18": "MuiPaper-elevation18-29",
              "elevation19": "MuiPaper-elevation19-30",
              "elevation2": "MuiPaper-elevation2-13",
              "elevation20": "MuiPaper-elevation20-31",
              "elevation21": "MuiPaper-elevation21-32",
              "elevation22": "MuiPaper-elevation22-33",
              "elevation23": "MuiPaper-elevation23-34",
              "elevation24": "MuiPaper-elevation24-35",
              "elevation3": "MuiPaper-elevation3-14",
              "elevation4": "MuiPaper-elevation4-15",
              "elevation5": "MuiPaper-elevation5-16",
              "elevation6": "MuiPaper-elevation6-17",
              "elevation7": "MuiPaper-elevation7-18",
              "elevation8": "MuiPaper-elevation8-19",
              "elevation9": "MuiPaper-elevation9-20",
              "root": "MuiPaper-root-9",
              "rounded": "MuiPaper-rounded-10",
            }
          }
          component="header"
          elevation={4}
          square={true}
        >
          <header
            className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
          >
            <WithStyles(Toolbar)>
              <Toolbar
                classes={
                  Object {
                    "gutters": "MuiToolbar-gutters-37",
                    "root": "MuiToolbar-root-36",
                  }
                }
                disableGutters={false}
              >
                <div
                  className="MuiToolbar-root-36 MuiToolbar-gutters-37"
                >
                  <WithStyles(Typography)
                    variant="title"
                  >
                    <Typography
                      align="inherit"
                      classes={
                        Object {
                          "alignCenter": "MuiTypography-alignCenter-83",
                          "alignJustify": "MuiTypography-alignJustify-85",
                          "alignLeft": "MuiTypography-alignLeft-82",
                          "alignRight": "MuiTypography-alignRight-84",
                          "body1": "MuiTypography-body1-79",
                          "body2": "MuiTypography-body2-78",
                          "button": "MuiTypography-button-81",
                          "caption": "MuiTypography-caption-80",
                          "colorError": "MuiTypography-colorError-93",
                          "colorInherit": "MuiTypography-colorInherit-89",
                          "colorPrimary": "MuiTypography-colorPrimary-90",
                          "colorSecondary": "MuiTypography-colorSecondary-91",
                          "colorTextSecondary": "MuiTypography-colorTextSecondary-92",
                          "display1": "MuiTypography-display1-74",
                          "display2": "MuiTypography-display2-73",
                          "display3": "MuiTypography-display3-72",
                          "display4": "MuiTypography-display4-71",
                          "gutterBottom": "MuiTypography-gutterBottom-87",
                          "headline": "MuiTypography-headline-75",
                          "noWrap": "MuiTypography-noWrap-86",
                          "paragraph": "MuiTypography-paragraph-88",
                          "root": "MuiTypography-root-70",
                          "subheading": "MuiTypography-subheading-77",
                          "title": "MuiTypography-title-76",
                        }
                      }
                      color="default"
                      gutterBottom={false}
                      headlineMapping={
                        Object {
                          "body1": "p",
                          "body2": "aside",
                          "display1": "h1",
                          "display2": "h1",
                          "display3": "h1",
                          "display4": "h1",
                          "headline": "h1",
                          "subheading": "h3",
                          "title": "h2",
                        }
                      }
                      noWrap={false}
                      paragraph={false}
                      variant="title"
                    >
                      <h2
                        className="MuiTypography-root-70 MuiTypography-title-76"
                      >
                        Movie Manager
                      </h2>
                    </Typography>
                  </WithStyles(Typography)>
                </div>
              </Toolbar>
            </WithStyles(Toolbar)>
          </header>
        </Paper>
      </WithStyles(Paper)>
    </AppBar>
  </WithStyles(AppBar)>
</Component>
`;
